sqs/main.tf

```diff
@@ -64,20 +64,42 @@ resource "aws_sqs_queue" "dlq_with_sse" {
   sqs_managed_sse_enabled   = true
 }
 
-module "dlq_cloudwatch_alarm" {
-  source              = "../cloudwatch_alarms"
-  metric_name         = "ApproximateNumberOfMessagesVisible"
-  namespace           = "AWS/SQS"
-  name                = "${var.queue_name}-messages-visible--dlq-alarm"
-  threshold           = var.dlq_cloudwatch_alarm_visible_messages_threshold
+resource "aws_cloudwatch_metric_alarm" "dlq_metric_alarm" {
+  alarm_name          = "${var.queue_name}-messages-visible--dlq-alarm"
   comparison_operator = "GreaterThanThreshold"
-  statistic           = "Sum"
-  treat_missing_data  = "ignore"
-  datapoints_to_alarm = 1
-  dimensions = {
-    QueueName = local.sqs_dlq.name
+  evaluation_periods  = 60
+  metric_query {
+    id = "m1"
+    metric {
+      dimensions = {
+        QueueName = "${var.queue_name}-dlq"
+      }
+      metric_name = "ApproximateNumberOfMessagesVisible"
+      period      = 60
```

Comment:
Just to add, should/could these `period`s be added to a variable?

Reply:
Yeah, can do. I've added a single variable as I can't see why we'd want different evaluation periods for the different metrics. I've defaulted them to 60 as this seems sensible.